File: e1/e1.rs
// Project euler 1

fn e1(n: int) -> int {
	match n {
		0 => 0,
		n => { if n%3 == 0 || n%5 == 0 {
				e1(n-1) + n
			} 
			else {
				e1(n-1)
			}
		}
	}
}
fn main() {
    io::println(fmt!("%d", e1(999)))
}

File: haskell/e1.hs
-- Project Euler 1

-- fkn
e1 n = sum [ x | x <- [1..n], x `mod` 5 == 0 || x `mod` 3 == 0]
-- hard coded
e1_999 = sum [ x | x <- [1..999], mod x 5 == 0 || mod x 3 == 0]
